src/contrib/cluster/Akka.Cluster.Tools/Singleton/ClusterSingletonManager.cs

```diff
@@ -199,119 +173,117 @@ internal sealed class YoungerData : IClusterSingletonData
         /// <summary>
         /// TBD
         /// </summary>
-        public List<UniqueAddress> Oldest { get; }
+        public ImmutableList<UniqueAddress> Oldest { get; }
 
         /// <summary>
         /// TBD
         /// </summary>
         /// <param name="oldest">TBD</param>
-        public YoungerData(List<UniqueAddress> oldest)
+        public YoungerData(ImmutableList<UniqueAddress> oldest)
         {
             Oldest = oldest;
         }
     }
 
     /// <summary>
-    /// TBD
+    /// State when we're transitioning to becoming the oldest singleton manager.
     /// </summary>
     [Serializable]
     internal sealed class BecomingOldestData : IClusterSingletonData
     {
         /// <summary>
-        /// TBD
+        /// The previous oldest nodes - can be empty
         /// </summary>
-        public List<UniqueAddress> PreviousOldest { get; }
-
-        /// <summary>
-        /// TBD
-        /// </summary>
-        /// <param name="previousOldest">TBD</param>
-        public BecomingOldestData(List<UniqueAddress> previousOldest)
+        public ImmutableList<UniqueAddress> PreviousOldest { get; }
+        
+        public BecomingOldestData(ImmutableList<UniqueAddress> previousOldest)
```

Comment:
Moved to using `Immutable` lists everywhere for all states and messages. These are only sent when movement is occurring in the cluster, which is inherently low-volume, so there's no performance concerns.